orchagent/intfsorch.cpp

```diff
@@ -206,6 +206,51 @@ bool IntfsOrch::setIntf(const string& alias, sai_object_id_t vrf_id, const IpPre
     return true;
 }
 
+bool IntfsOrch::removeIntf(const string& alias, sai_object_id_t vrf_id, const IpPrefix *ip_prefix)
+{
+    SWSS_LOG_ENTER();
+
+    Port port;
+    if (!gPortsOrch->getPort(alias, port))
+    {
+        return false;
+    }
+
+    if (m_syncdIntfses.find(alias) == m_syncdIntfses.end())
+    {
+        return false;
+    }
+
+    if (ip_prefix && m_syncdIntfses[alias].ip_addresses.count(*ip_prefix))
+    {
+        removeSubnetRoute(port, *ip_prefix);
+        removeIp2MeRoute(vrf_id, *ip_prefix);
+
+        if(port.m_type == Port::VLAN)
+        {
+            removeDirectedBroadcast(port, *ip_prefix);
+        }
+
+        m_syncdIntfses[alias].ip_addresses.erase(*ip_prefix);
+    }
+
+    /* Remove router interface that no IP addresses are associated with */
+    if (m_syncdIntfses[alias].ip_addresses.size() == 0)
```

Comment:
Looks like for VRF implementation, this `if-check` handles the scenario where we get delete of interface without ip_prefix first and later comes the delete of interface with ip_prefix. From the below code, looks like this sequence is handled for Bitmap implementation also. Can you please confirm?

Reply:
Yes, this sequence is handled for Bitmap implementation (since remove flow is different). Once it is removed in Bitmap ```removeIntf()``` we also  call this ```IntfsOrch::removeIntf``` function but with ip_prefix set to ```nullptr``` so it allows to remove all left SAI object.